test/e2e/authorization.go

```diff
@@ -0,0 +1,572 @@
+				}
+				g.It("should allow read access", func() {
+					tc.data.verbs = []string{"get", "list", "watch"}
+					tc.run(context.TODO(), cs, true)
+					gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+				})
+				g.It("should deny write access", func() {
+					tc.data.verbs = []string{"create", "update", "delete", "patch"}
+					tc.run(context.TODO(), cs, false)
+					gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+				})
+			})
+		})
+	})
+
+	g.Context("For PowerUser, Manual and Emergency groups", func() {
+		var tc testCase
+		g.BeforeEach(func() {
+			tc.data.groups = [][]string{
+				{"PowerUser"},
+				{"Manual"},
+				{"Emergency"},
+			}
+			tc.data.users = []string{"test-user"}
+		})
+
+		g.It("should deny read access to Secrets in kube-system and visibility namespaces", func() {
+			tc.data.resources = []string{"secrets"}
+			tc.data.namespaces = []string{"kube-system", "visibility"}
+			tc.data.verbs = []string{"get", "list", "watch"}
+			tc.run(context.TODO(), cs, false)
+			gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+		})
+
+		g.It("should deny write access to Nodes", func() {
+			tc.data.resources = []string{"nodes"}
+			tc.data.verbs = []string{"create", "update", "delete", "patch"}
+			tc.run(context.TODO(), cs, false)
+			gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+		})
+
+		g.It("should deny write access to DaemonSets", func() {
+			tc.data.resources = []string{"apps/daemonsets"}
+			tc.data.verbs = []string{"create", "update", "delete", "patch"}
+			tc.run(context.TODO(), cs, false)
+			gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+		})
+
+		g.It("should allow deleting CRDs", func() {
+			tc.data.resources = []string{"apiextensions.k8s.io/customresourcedefinitions"}
+			tc.data.verbs = []string{"delete"}
+			tc.run(context.TODO(), cs, true)
+			gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+		})
+
+		g.It("should deny deleting kube-system or visibility namespaces", func() {
+			tc.data.resources = []string{"namespaces"}
+			tc.data.namespaces = []string{"kube-system", "visibility"}
+			tc.data.verbs = []string{"delete"}
+			tc.run(context.TODO(), cs, false)
+			gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+		})
+
+		g.When("the resource is a namespaced resource", func() {
+			g.BeforeEach(func() {
+				tc.data.resources = []string{
+					"pods",
+					"apps/deployments",
+					"apps/statefulsets",
+					"apps/deployments/scale",
+					"apps/statefulsets/scale",
+					"services",
+					"persistentvolumes",
+					"persistentvolumeclaims",
+					"configmaps",
+				}
+				tc.data.verbs = []string{"create", "update", "delete", "patch"}
+			})
+			// These should be covered by the admission-controller tests.
+			// They're written here for completeness.
+			g.It("should deny write access in kube-system and visibility namespaces", func() {})
```

Comment:
Can we implement them but `skip` them? This way we can validate them if we later manage to implement them with RBAC.

Reply:
Sounds good. I'll do it.

Reply:
Done in this commit: https://github.com/zalando-incubator/kubernetes-on-aws/pull/8536/commits/6f7f8c96c5ef8339c41832b8ffdc86a8a9b46b97

I found another write protection test that failed so I skipped that as well, and marked it with a comment in the code.